app/controllers/posts_controller.rb

```diff
@@ -27,6 +27,17 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+
+    if can? :destroy, @post
+      @post.destroy
+      redirect_to "/users/#{current_user.id}/posts", notice: 'Successfully deleted.'
+    else
+      redirect_to user_post_path(@post.author_id, @post), alert: 'Unauthorized action.'
+    end
+  end
+
   private
 
   def post_params
```

Comment:
- [ ] It will be great to ensure that in your PostsController you add` load_and_authorize_resource` method at the top. This is provided to automatically authorize all actions in a RESTful-style resource controller. It will use a before action to load the resource into an instance variable and authorize it for every action. 

### EXAMPLE:

```ruby
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

Reply:
- [ ] Well resolved 🥇